### PR TITLE
ENH: function to verify FS license

### LINKED ISSUE #43: Create a FreeSurfer-license canary that can be run by fMRIPrep
https://openneuro.org/datasets/ds001230/versions/00001?app=FMRIPREP&version=51&job=5a847e689259cd00018b9810

Possible introduced in 1.0.7

`--n_cpus 4 --participant_label 3129 --use-aroma --fs-license-file /input/data/b39610adfe02fdce2ad9d30797c567b3/LICENSE --output-space template T1w fsaverage`

```
[Node] Error on "fmriprep_wf.single_subject_3129_wf.anat_preproc_wf.surface_recon_wf.autorecon1" (/root/src/fmriprep/work/fmriprep_wf/single_subject_3129_wf/anat_preproc_wf/surface_recon_wf/autorecon1)
Traceback (most recent call last):
  File "/usr/local/miniconda/bin/fmriprep", line 11, in <module>
    load_entry_point('fmriprep==1.0.7', 'console_scripts', 'fmriprep')()
  File "/usr/local/miniconda/lib/python3.6/site-packages/fmriprep/cli/run.py", line 267, in main
    fmriprep_wf.run(**plugin_settings)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/engine/workflows.py", line 602, in run
    runner.run(execgraph, updatehash=updatehash, config=self.config)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/plugins/base.py", line 168, in run
    self._clean_queue(jobid, graph, result=result))
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/plugins/base.py", line 227, in _clean_queue
    raise RuntimeError("".join(result['traceback']))
RuntimeError: Traceback (most recent call last):
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/plugins/multiproc.py", line 68, in run_node
    result['result'] = node.run(updatehash=updatehash)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/engine/nodes.py", line 487, in run
    result = self._run_interface(execute=True)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/engine/nodes.py", line 571, in _run_interface
    return self._run_command(execute)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/engine/nodes.py", line 650, in _run_command
    result = self._interface.run(cwd=outdir)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/interfaces/base/core.py", line 516, in run
    runtime = self._run_interface(runtime)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/interfaces/base/core.py", line 1023, in _run_interface
    self.raise_exception(runtime)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/interfaces/base/core.py", line 960, in raise_exception
    ).format(**runtime.dictcopy()))
RuntimeError: Command:
recon-all -autorecon1 -i /snapshot/data/sub-3129/anat/sub-3129_acq-MPRAGE_run-1_T1w.nii.gz -noskullstrip -openmp 3 -subjid sub-3129 -sd /output/data/freesurfer
Standard output:
ERROR: You are trying to re-run an existing subject with (possibly)
 new input data (-i). If this is truly new input data, you should delete
 the subject folder and re-run, or specify a different subject name.
 If you are just continuing an analysis of an existing subject, then 
 omit all -i flags.
Standard error:

Return code: 1
```

### PR DESCRIPTION
**Pull request by mgxd**
Closes #43

### CODE REVIEW

File: smriprep/utils/misc.py [hunk 1]
```diff
@@ -21,3 +21,36 @@ def apply_lut(in_dseg, lut, newpath=None):
...
+    iflogger = logging.getLogger('nipype.interface')
+    _level = iflogger.level
+    iflogger.setLevel(0)  # mute logging
+
+    tmp_file = 'test.nii.gz'
+    out_file = 'out.mgz'
+    # create test NIfTI
+    nb.Nifti1Image(np.zeros((5, 5, 5)), np.eye(4)).to_filename(tmp_file)
+    # run simple fs interface
+    mric = MRIConvert(in_file=tmp_file, out_file=out_file, out_type='mgz')
```
Comment (oesteban) [inline smriprep/utils/misc.py:46]: Why not subprocess straightaway? That would rid you of changing the logger, which may have some other unexpected side effects.

File: smriprep/utils/misc.py [hunk 2]
```diff
@@ -21,3 +21,36 @@ def apply_lut(in_dseg, lut, newpath=None):
...
+    res = mric.run(ignore_exception=True)
+    valid = "ERROR: FreeSurfer license file" not in res.runtime.stderr
+    # clean up
+    with contextlib.suppress(FileNotFoundError):
```
Comment (oesteban) [inline smriprep/utils/misc.py:50]: Although I like this context, why not use a `tempfile.TemporaryFolder` context instead?

File: smriprep/utils/misc.py [hunk 3]
```diff
@@ -21,3 +21,35 @@ def apply_lut(in_dseg, lut, newpath=None):
                    segm.affine, hdr).to_filename(out_file)
 
     return out_file
+
+
+def check_valid_fs_license():
```
Comment (oesteban) [inline smriprep/utils/misc.py:26]: Are you planning on moving this function definition (`check_valid_fs_license`) to niworkflows? I think that was what I had in mind in first place. s/fMRIPrep will only need a quick call to it.

**Comment by mgxd** [comment]
sure, i'll open a PR there.

**Pull request closed without merging**